torch/_library/infer_schema.py

```diff
@@ -53,6 +53,13 @@ def infer_schema(
         (Tensor x) -> Tensor
     """
     UNKNOWN_MUTATES = "unknown"
+    pf_globals = prototype_function.__globals__
+    # To do this properly we really need `locals()` from the scope of the
+    # prototype_function. Unfortunately there doesn't seem to be any way to get
+    # that consistently.
+    pf_locals = None
+    # TODO: py3.10+ pass the `globals` parameter and we no longer need to deal
+    # with stringified annotations.
```

Comment:
What do you mean?

Reply:
As of py3.10+ inspect.signature() accepts a `globals` (and `locals`) parameter which if specified will do the eval() within the `inspect.signature` itself so it will never return stringified types.